lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -109,8 +115,9 @@ sub results {
         my $text_file_name = $step->{text};
         if (!$skip_text_data && $text_file_name && !defined $step->{text_data}) {
             my $text_file = path($dir, $text_file_name);
-            my $text_data = eval { decode('UTF-8', $text_file->slurp) };
-            $step->{text_data} = $text_data // "Unable to read $text_file_name.";
+            my $text_data;
+            try { $step->{text_data} = decode('UTF-8', $text_file->slurp) }
+            catch ($e) { $step->{text_data} = "Unable to read $text_file_name." }
```

Comment:
Here you can actually use the `do` construct

```suggestion
            $step->{text_data} = do {
                try { decode('UTF-8', $text_file->slurp) }
                catch ($e) { "Unable to read $text_file_name." }
            };
```

Reply:
Same question I had for Tina.
In additional Tina didnt comment on that (like she approved the change) but you want it with the do block. is there a conflict or any good reason?

Reply:
I just didn't notice it. I just saw the unused variable and didn't look about if the rest could be simplified more